x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterLookupJoinIT.java

```diff
@@ -374,6 +374,34 @@ public void testLookupJoinMissingKey() throws IOException {
         );
         assertThat(ex.getMessage(), containsString("Unknown column [local_tag] in right side of join"));
 
+        // Add KEEP clause to try and trick the field-caps result parser
```

Comment:
Could you please explain why this is needed?

Reply:
Because when `KEEP` is present, field-caps call only fetches the restricted set of fields, and the code here: 
```
boolean allEmpty = true;
        for (FieldCapabilitiesIndexResponse ir : fieldCapsResponse.getIndexResponses()) {
            allEmpty &= ir.get().isEmpty();
        }
        // If all the mappings are empty we return an empty set of resolved indices to line up with QL
        var index = new EsIndex(indexPattern, rootFields, allEmpty ? Map.of() : concreteIndices, partiallyUnmappedFields);
``` 
in IndexResolver can produce empty mapping. Which hasn't been handled properly by the code because I assumed this could not happen.